### ISSUE TITLE
Add option "delegate to super" to @Delegate methods generator

### ISSUE THREAD

**Issue by remyvrs** [rzwitserloot/lombok#900]
Hi, 
I would really need this feature from this library :)
In case it is not already clear from the title I would need that the generated methods would be not in the form of "foo.add()" but in the form of "super.add()" and of course this could be done only for some explicit option given.

Thanks a lot in advance ! 
Regards

**Comment by remyvrs** [comment]
Hi,
I have patched the code according to my needs and made a patched release that I can use in my project. I will paste the patch bellow, so could you please add the feature ? 
Thank you!

diff --git a/src/core/lombok/experimental/Delegate.java b/src/core/lombok/experimental/Delegate.java
index d94389b..e803d7d 100644
--- a/src/core/lombok/experimental/Delegate.java
+++ b/src/core/lombok/experimental/Delegate.java
@@ -61,4 +61,10 @@
 	 * NB: All methods in {@code Object}, as well as {@code canEqual(Object other)} will never be delegated.
 	 */
 	Class<?>[] excludes() default {};
+	/**
+	 * Defines whether the calls are delegated to parent or not. 
+	 * Default value is false which means the calls are delegated to the field.
+	 * @return
+	 */
+	boolean delegatesToSuper() default false;
 }
diff --git a/src/core/lombok/javac/handlers/HandleDelegate.java b/src/core/lombok/javac/handlers/HandleDelegate.java
index fc3c81f..4666736 100644
--- a/src/core/lombok/javac/handlers/HandleDelegate.java
+++ b/src/core/lombok/javac/handlers/HandleDelegate.java
@@ -118,7 +118,7 @@
 				annotationNode.addError(LEGALITY_OF_DELEGATE);
 				return;
 			}
-			delegateReceiver = DelegateReceiver.FIELD;
+			delegateReceiver = "true".equals(annotation.getActualExpression("delegatesToSuper"))?DelegateReceiver.SUPER:DelegateReceiver.FIELD;
 			delegateType = member.type;
 			if (delegateType == null) reso.resolveClassMember(annotationNode.up());
 			delegateType = member.type;
@@ -433,6 +433,12 @@
 	}
 	
 	public enum DelegateReceiver {
+		SUPER {
+			public JCExpression get(final JavacNode node, final Name name) {
+				final JavacTreeMaker maker = node.getTreeMaker();
+				return maker.Ident(node.toName("super"));
+			}
+		},
 		METHOD {
 			public JCExpression get(final JavacNode node, final Name name) {
 				com.sun.tools.javac.util.List<JCExpression> nilExprs = com.sun.tools.javac.util.List.nil();

**Comment by sixcorners** [comment]
When would this be used? Doesn't the method in the super class get called by default if you don't override it in the subclass?

**Comment by remyvrs** [comment]
no, actually not. because if you generate an empty methos it will not call the super one
